Pull-Request_kotlin_Module_Work_Sprint_7

MaxSatin opened the pull request:
- Постарался выполнить основные условия задачи.
- Хотел дописать логику добавления Архивов внутрь Архивов, но не успеваю, за что прошу простить, пара функций поэтому не используется(
- Буду рад услышать и исправить все комментарии.

Заранее спасибо!

MaxSatin committed:
Add files via upload

MaxSatin committed:
Add files via upload

Second_Version_App

MaxSatin committed:
Add files via upload

Version_3_App

MaxSatin committed:
Add files via upload

Version_4_App

MaxSatin committed:
Add files via upload

Ver_5

nadzyathedude commented:
привет! из-за того, что ты создал новый пэкидж Kotlin-Module_Project, но оставил файл Main.kt, запускается именно он 
![image](https://github.com/MaxSatin/Kotlin-Module-Project/assets/62635172/fd62e97c-846d-4d9c-bfcb-1fca5f7fc5a1)

И выводит Hello World
![image](https://github.com/MaxSatin/Kotlin-Module-Project/assets/62635172/f1cfa9c0-6b7a-4011-88c9-56b28ab56b71)
 
А функцию для старта приложения с заметками невозможно запустить 
![image](https://github.com/MaxSatin/Kotlin-Module-Project/assets/62635172/5bb52e78-dc52-4de4-8b5c-4b3af0d70533)

nadzyathedude commented on Kotlin-Module-Project/src/main/kotlin/ArchiveManip.kt:
```diff
@@ -0,0 +1,45 @@
+import java.util.Scanner
+
+
+class ArchiveManip {
```
стоит подумать над неймингом класса. Можно оставить просто Archive. Зачастую не стоит сокращать слова в нейминге, так как бывает и не разберешь что же за слово подразумевалось

nadzyathedude commented on Kotlin-Module-Project/src/main/kotlin/JavaSDK.java:
```diff
@@ -0,0 +1,2 @@
+public class JavaSDK {
+}
```
кажется, этот файл не нужен

nadzyathedude commented on Kotlin-Module-Project/src/main/kotlin/ArchiveManip.kt:
```diff
@@ -24,22 +24,29 @@ class ArchiveManip {
                     when (readInput()) {
-                        "0" -> { createArchive(); break}
+                        "0" -> {
+                            createArchive(); break
```
после ; всегда стоит делать перенос строки 
```
createArchive();
 break
```

nadzyathedude commented on Kotlin-Module-Project/src/main/kotlin/Archive.kt:
```diff
@@ -2,49 +2,77 @@ class Archive (
-}
\ No newline at end of file
+
+        override fun toString(): String {
+            return "$name"
```
мне кажется эта функция не совсем имеет смысл, а переопределение настолько базовой функции toString здесь просто делает из нее другую функцию, а не расширяет ее функционал

MaxSatin replied:
А вот тут ты могла бы подсказать, пожалуйста. Я хотел, чтобы при вызове функции печаталось только имя объекта. Ты бы порекомендовала просто создать стороннюю функцию? 

Спасибо!

MaxSatin commented:
> привет! из-за того, что ты создал новый пэкидж Kotlin-Module_Project, но оставил файл Main.kt, запускается именно он ![image](https://private-user-images.githubusercontent.com/62635172/327383328-fd62e97c-846d-4d9c-bfcb-1fca5f7fc5a1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ2NDk5NDQsIm5iZiI6MTcxNDY0OTY0NCwicGF0aCI6Ii82MjYzNTE3Mi8zMjczODMzMjgtZmQ2MmU5N2MtODQ2ZC00ZDljLWJmY2ItMWZjYTVmN2ZjNWExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAyVDExMzQwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5ZDQ1MjljYWRkOTIzMGRkZTI0NGFhZTkwZjBhZjg1NTg4Y2IwYjE3NjA5NTg3NjRmZGJmZTU4MTUyMzNkYmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.43ABjeTv0L5U_KZpbjiDb2gm0L48doOF0M9_YqPRA20)
> 
> И выводит Hello World ![image](https://private-user-images.githubusercontent.com/62635172/327383023-f1cfa9c0-6b7a-4011-88c9-56b28ab56b71.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ2NDk5NDQsIm5iZiI6MTcxNDY0OTY0NCwicGF0aCI6Ii82MjYzNTE3Mi8zMjczODMwMjMtZjFjZmE5YzAtNmI3YS00MDExLTg4YzktNTZiMjhhYjU2YjcxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAyVDExMzQwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMwMjA2ODhhNTVmZGE0NGRkZmVkNDk5ZDczOGYxZTA5ZTMyYWMxNGY4ODhmMmI4Y2EyM2NkYzcwMmRkYTEzOTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ilAV1KDvZCt2Va-HYFAXs0eXPOAMPZcgKyL8xWttsRo)
> 
> А функцию для старта приложения с заметками невозможно запустить ![image](https://private-user-images.githubusercontent.com/62635172/327383162-5bb52e78-dc52-4de4-8b5c-4b3af0d70533.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ2NDk5NDQsIm5iZiI6MTcxNDY0OTY0NCwicGF0aCI6Ii82MjYzNTE3Mi8zMjczODMxNjItNWJiNTJlNzgtZGM1Mi00ZGU0LThiNWMtNGIzYWYwZDcwNTMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAyVDExMzQwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc0YzJmMTlmY2UyZGU4MGFiMDZhOTBlMjE4NjhiMmM2ZDg1YmRjOGQ5MjI3YWJiMDhlMmI2OGU3YWU0ODU5ZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DC9OOEvWafm-DtzFrh3bfB4384OqQm2mPt9VZz_0pwI)

Надя, привет! 
Спасибо за комментарии! Я только не совсем понимаю, как так получилось. На моем компьютере файл открывается только с 1 Main файлом. Вот тут скрин. Ты не посмотришь, пожалуйста? 

![image](https://github.com/MaxSatin/Kotlin-Module-Project/assets/148345998/bf9ac88f-86e9-420b-8809-7046a651134a)

MaxSatin committed:
Final_version_reAdded_02/05

MaxSatin committed:
Final_version_reAdded_02/05 21:17
Правки:

Название класса ArchiveManip изменено на ArchiveManipulator
Добавлена логика проверки пользовательского ввода при выборе Архива из списка

Исправлены синтаксические рекомендации

nadzyathedude commented:
привет! проверила твою работу, сейчас все работает как надо и файлы исправно запускаются. честно говоря не знаю что могло случится в первый раз с файлами, могу лишь предположить что что-то наплодил гит
